esm_parser/esm_parser.py

```diff
@@ -621,11 +621,17 @@ def dict_merge(dct, merge_dct):
                         dct["debug_info"]["loaded_from_file"] = [dct["debug_info"]["loaded_from_file"]]
                     else:
                         dct["debug_info"]["loaded_from_file"].append(merge_dct["debug_info"]["loaded_from_file"])
+        # MA: I'm not super happy about the resolve_nested_adds implementation. Nested
+        # adds should probably resolved in a different place, after the first level
+        # ones are resolved.
         # If the key exists and starts by ``add_``, it is a nested ``add_`` and is
         # solved as such
         elif (
-            isinstance(k, str)
+            resolve_nested_adds
+            and isinstance(k, str)
             and k.startswith("add_")
+            and "module_actions" not in k
```

Comment:
I'm not really happy about these two lines. Can we save specific keys to skip externally somehow? Here it is hardcoded. I mean, externally is just moving the problem elsewhere...so maybe it doesn't matter.

Reply:
Yes, maybe something like 
```python
protected_items = ["module_actions", "foo", "bar"]
for item in protected items:
    if item not in k:
        ...
```
Something like this, right?

Reply:
I agree with what you are both saying.

I don't love the idea of an external file here, for the cross dependencies potential issues, but some sets configuration variables are already defined internally in the esm_parser, maybe that's a good enough solution?

Reply:
yeah, just pull it out of the loop. I would recommend we discuss the "mono-repo" idea at our mini-retreat, I feel this would make many things a bit easier.

Reply:
Just made the changes for making it more general. I am happier now. Thanks a lot for the suggestions @pgierz and @denizural !